src/appleDeviceUtils/appleDeviceUtils.js

```diff
@@ -0,0 +1,102 @@
+// Do not remove -- Apple devices change this when they support modern webkit messaging
+let hasModernWebkitAPI = false
+// INJECT hasModernWebkitAPI HERE
+
+// The native layer will inject a randomised secret here and use it to verify the origin
+let secret = 'PLACEHOLDER_SECRET'
+
+const ddgGlobals = require('./captureDdgGlobals')
+
+/**
+ * Sends message to the webkit layer (fire and forget)
+ * @param {String} handler
+ * @param {*} data
+ * @returns {*}
+ */
+const wkSend = (handler, data = {}) =>
+    window.webkit.messageHandlers[handler].postMessage({...data, messageHandling: {...data.messageHandling, secret}})
+
+/**
+ * Generate a random method name and adds it to the global scope
+ * The native layer will use this method to send the response
+ * @param {String} randomMethodName
+ * @param {Function} callback
+ */
+const generateRandomMethod = (randomMethodName, callback) => {
+    Object.defineProperty(ddgGlobals.window, randomMethodName, {
```

Comment:
This can be hooked here.
```suggestion
    ddgGlobals.ObjectDefineProperty(ddgGlobals.window, randomMethodName, {
```